chia/wallet/trading/offer.py

```diff
@@ -72,7 +72,7 @@ def from_condition_and_nonce(cls, condition: Program, nonce: bytes32) -> Notariz
         return cls(puzzle_hash, amount, memos, nonce)
 
 
-@dataclass(frozen=True)
+@dataclass(frozen=True, eq=False)
```

Comment:
all changes in this file appear unrelated. Are they a merge mistake?

Reply:
let me look back at commit history

Reply:
it was to fix a comparison bug between offers for the test: https://github.com/Chia-Network/chia-blockchain/commit/229934fe2ebf2f03462848b1d824dac9feb40fd6

Reply:
so we still have this problem in `main`? It would be great to land this as its own PR, ideally with a test demonstrating the problem

Reply:
i looked into it, its so that the cache isnt printed when comparing as sometimes the offer is just printed, and the order could be different in the cache so it would make the test fail